src/main/scala/extractors/Extractor.scala

```diff
@@ -0,0 +1,50 @@
+package extractors
+
+object Extractor {
+
+  val inUaStr: String = "in.ua"
+  val gmailComStr: String = "gmail.com"
+
+  def filterNotInUa(domains: List[String]): List[String] = {
+    domains.filterNot {
+      case Domain(f, s, _*) if s"$s.$f" == inUaStr => true
+      case _ => false
+    }
+  }
+
+  def filterGmail(emails: Array[String]): Array[Unit] = {
+    for (email <- emails) yield {
+      email match {
+        case Email(_, domain) if domain == gmailComStr => println(email)
+        case _ =>
+      }
+    }
+  }
+
+//  def filterGmail(emails: Array[String]) = {
+//    for {
+//      emailStr <- emails
+//      email <- Email.unapply(emailStr) if email._2 == gmailComStr
+//    } yield (println(emailStr))
```

Comment:
Using variable decomposing instead of `unapply` method call will look more natural.
```suggestion
    for {
      emailStr <- emails
      Email(_, domain) = emailStr if domain == gmailComStr
    } yield (println(emailStr))
```

Reply:
Please, try not to use `._1`  tuple methods as they are not readable. Instead use variable decomposition.

Reply:
Got it. This commented code is already removed according to 1st comment, but will use this in future